pkg/dynamicaccess/diffieHellman.go

```diff
@@ -0,0 +1,31 @@
+package dynamicaccess
+
+import (
+	"crypto/ecdsa"
+	"errors"
+
+	"github.com/ethersphere/bee/pkg/crypto"
+)
+
+type DiffieHellman interface {
+	SharedSecret(publicKey *ecdsa.PublicKey, tag string, moment []byte) ([]byte, error) // tag- topic?
+}
```

Comment:
Create an interface that is called `Session`.
Session would have functions: `New` and `Key`.
Diffie-Hellman is an expensive operation, because user input is required.
In AccessLogic, only one Diffie-Hellman should happen, so in Diffie-Hellman, this operation should happen at init.

That's how it should look like from the outside:
`sessionInstance = Session.New(keystore, tag)`
key = sessionInstance.Key(publicKey, ...nonce) // list of moments, optional


First, initialize with private key
Second, ask with specific public key
Third, ask moments, for the given public key.
// Viktor

Reply:
Done.
Password protection is only mocked as of now, will be implemented later.